### PR TITLE
Fix lifted nullable binary operator semantics

### PR DESCRIPTION
**Pull request by Jim8y**
## Summary
- preserve null propagation for lifted arithmetic and bitwise operators
- implement C# three-valued semantics for nullable Boolean `&` and `|`
- apply the same behavior to compound assignments without changing evaluation order
- handle lifted `BigInteger` operations correctly

## Tests
- added coverage for integral types, nullable enums, `BigInteger`, Boolean operators, compound assignments, evaluation order, overflow checks, and both optimization modes
- passed all 1,542 compiler unit tests

### CODE REVIEW

**Review by shargon — DISMISSED** [review]

**Review by ajara87 — DISMISSED** [review]

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/BinaryExpression.cs [hunk 1]
```diff
@@ -150,6 +165,120 @@ type is INamedTypeSymbol
...
+        LdLoc(leftSlot);
+        IsNull();
+        JumpIfTrue(nullTarget);
+        LdLoc(rightSlot);
+        IsNull();
+        JumpIfTrue(nullTarget);
+
+        LdLoc(leftSlot);
+        LdLoc(rightSlot);
+        EmitBinaryOperator(model, expression, model.GetTypeInfo(expression).Type!);
```
Comment (cschuchardt88) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/BinaryExpression.cs:229]: **[bug]** Lifted `bool? ^ bool?` goes through `ConvertLiftedBinaryExpression` (operator `^` is in the lifted set; the three-valued path is only `&`/`|`). The emit type is `bool?`. `GetStackItemType` does not unwrap `Nullable<T>`, so `bool?` becomes `StackItemType.Any` and `EmitBinaryOperator` skips the XOR→Boolean `Nz()` fix. NeoVM `XOR` yields Integer 0/1. Direct `BoolXor` tests still pass via engine coercion; `(true ^ true) & true` becomes `true` because Integer 0 is not `EQUAL` to `PUSHF`. Same hole for `^=`.

**Suggestion:** Unwrap `Nullable<T>` before `GetStackItemType` / `isBoolean`. Pass underlying `bool` into `EmitBinaryOperator` so `Nz()` runs. Add tests for `(true ^ true) & true`, `(false ^ false) | false`, and `bool? ^= bool?` with both non-null.

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/BinaryExpression.cs [hunk 2]
```diff
@@ -150,6 +165,120 @@ type is INamedTypeSymbol
...
+        var decisiveTarget = new JumpTarget();
+        var nullTarget = new JumpTarget();
+        var endTarget = new JumpTarget();
+
+        LdLoc(leftSlot);
+        if (isAnd)
+            PushF();
+        else
+            PushT();
+        AddInstruction(OpCode.EQUAL);
```
Comment (cschuchardt88) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/BinaryExpression.cs]: **[suggestion]** Three-valued `&`/`|` detect a decisive operand with `EQUAL` against `PUSHF`/`PUSHT`. That is correct for ABI/literal `bool?` values (Boolean stack items). It is not correct for Integer 0/1, which `JMPIF`/`JMPIFNOT` would treat as false/true. Combined with the XOR issue, any Integer-typed `bool?` is neither decisive nor null and takes the fallthrough. Tests only cover null mixed with a Boolean, plus `BoolXor` returning to the engine.

**Suggestion:** After `ISNULL`, use `JMPIFNOT` (AND) / `JMPIF` (OR). Extend tests with the full 3×3 tables for `&`/`|`/`^` (including both-non-null) and mixed `bool & bool?`.

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]

**Review by Wi1l-B0t — DISMISSED** [review]

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.ComplexAssignment.cs [hunk 3]
```diff
@@ -274,8 +274,56 @@ private void EmitComplexAssignmentOperator(SemanticModel model, ITypeSymbol type
...
+        LdLoc(leftSlot);
+        LdLoc(rightSlot);
+        EmitComplexAssignmentOperatorCore(model, type, operatorToken, right, type.GetStackItemType());
+        Jump(OpCode.JMP_L, endTarget);
+
+        nullTarget.Instruction = PushNull();
+        endTarget.Instruction = Nop();
+    }
+
+    private void EmitComplexAssignmentOperatorCore(SemanticModel model, ITypeSymbol type, SyntaxToken operatorToken, ExpressionSyntax right, VM.Types.StackItemType itemType)
```
Comment (cschuchardt88) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.ComplexAssignment.cs]: **[nit]** `EmitComplexAssignmentOperatorCore` takes `itemType` and immediately overwrites it with `GetStackItemType(GetNonNullableValueType(type))`. Callers therefore pass a meaningless value (`type.GetStackItemType()` on `int?`/`bool?` is `Any`, not the underlying integer/boolean). Harmless, but the extra parameter is dead after the unwrap that this PR added.

**Suggestion:** Drop the `itemType` parameter and compute it once after `GetNonNullableValueType`.

**Review by cschuchardt88 — APPROVED** [review]

**Review by ajara87 — APPROVED** [review]

**Review by Wi1l-B0t — APPROVED** [review]
